### ISSUE TITLE
Displaying a diff from nodegit: should I try to bypass diff2html's parse

### ISSUE THREAD

**Issue by julien-c** [rtfpessoa/diff2html#344]
Hi,

I am displaying a diff from a git commit, using nodegit – basically following [this official example](https://github.com/nodegit/nodegit/blob/master/examples/diff-commits.js).

I'm wondering if it's ok, or "good enough", to serialize the diff to a string representation, and then to have `diff2html` parse it again, even though we have the structured data (i.e., some `DiffList`, `Diff`, `Patch`, `Hunk`, and `DiffLine`s objects) in nodegit's output already.

In other words, should I try to transform my diff data into `diff2html`'s `DiffFile[]` and bypass the `.parse`?

[Of course `diff2html`'s `DiffFile[]` might be tedious to generate, so maybe I shouldn't worry about it...]

What do you think? Thank you!

**Comment by rtfpessoa** [comment]
Hi @julien-c,
I think the best way is definitely for you to get the string representation of the diff like you would get out of a git diff or the unix diff cli. That way you are sure it will work fine.

Mapping the structured data is going to be more work on you.

**Comment by julien-c** [comment]
Makes sense. Thank you! cc @pierrci
